### PR TITLE
Support for TIMESTAMPTZ

### LINKED ISSUE #228: Support timestamptz type
Generally it's recommended to use the `timestamptz` type instead of the `timestamp` type, so we should support this. I think supporting this is probably very simple as it's stored the same on disk as the `timestamp` type, but it needs a few tests too.

### PR DESCRIPTION
**Pull request by mkaruza**
Fixes #228

### CODE REVIEW

File: test/regression/sql/type_support.sql [hunk 1]
```diff
@@ -131,10 +148,10 @@ SELECT * FROM uuid_tbl;
...
 CREATE TABLE json_tbl(a JSON);
 INSERT INTO json_tbl SELECT CAST(a as JSON) FROM (VALUES
-    ('{"key1": "value1", "key2": "value2"}'),
-    ('["item1", "item2", "item3"]'),
-    (NULL),
-    ('{}')
+	('{"key1": "value1", "key2": "value2"}'),
+	('["item1", "item2", "item3"]'),
+	(NULL),
+	('{}')
```
Comment (JelteF) [inline test/regression/sql/type_support.sql:154]: nit: according to our `.editorconfig` we use spaces for indentation in our SQL files. (same with the indentation above)

File: test/regression/expected/type_support.out [hunk 2]
```diff
@@ -132,6 +132,46 @@ SELECT * FROM timestamp_tbl;
  Mon May 15 12:30:45 2023
 (3 rows)
 
+-- TIMESTAMP_TZ
+SET DateStyle = 'ISO, YMD';
```
Comment (JelteF) [inline test/regression/expected/type_support.out:136]: Let's wait for #293 so we can remove this `SET`.

**Review by JelteF — DISMISSED** [review]
Overall LGTM, but let's wait for #293

**Comment by mkaruza** [comment]
Rebased on top of #293

File: test/regression/sql/type_support.sql (hunk 1 above)
Comment (mkaruza) [reply test/regression/sql/type_support.sql]: Converted whole sql file to space indentation.

**Review by JelteF — APPROVED** [review]

**Comment by JelteF** [comment]
Needs a rebase now that #293 was merged

**Review by JelteF — APPROVED** [review]

**Review by wuputah — APPROVED** [review]

**Pull request merged**